hazelcast/src/main/java/com/hazelcast/jet/impl/MasterJobContext.java

```diff
@@ -1000,27 +1016,59 @@ boolean maybeScaleUp(int dataMembersWithPartitionsCount) {
         return false;
     }
 
-    List<RawJobMetrics> jobMetrics() {
-        return jobMetrics;
+    /**
+     * Aggregated job and execution metrics.
+     * @see #metrics
+     */
+    List<RawJobMetrics> metrics() {
+        return metrics;
+    }
+
+    /**
+     * Generates persistent job-specific metrics —currently only job status.
+     *
+     * @param status the advertised status
+     * @implNote
+     * Since {@linkplain JobExecutionService#beginExecution0 terminal metrics are collected}
+     * before {@linkplain #finalizeExecution execution is finalized}, the last reported job
+     * status is not terminal. It is also not possible to infer terminal metrics collection
+     * cycle, so that the correct status is advertised. As a solution, the job status metric
+     * is generated on demand.
+     *
+     * @see MasterContext#provideDynamicMetrics
+     */
+    private void setJobMetrics(JobStatus status) {
+        if (!mc.metricsEnabled()) {
+            return;
+        }
+        MetricsCompressor compressor = new MetricsCompressor();
```

Comment:
`setJobMetrics` seems to be invoked in each collection cycle when the job is suspended (so MC exists). Creation of MetricsCompressor is rather costly and in other places we try to not create it too often. OTOH, it have to be closed deterministically to not leak native resources under load.

Reply:
A better solution is to collect status metric only on status change, which occurs much less frequent than metrics collection cycles. And, `MetricsCompressor.getBlobAndClose()` calls `close()` internally.

Reply:
@Fly-Style `MetricsCompressor` is currently the only way to convert metric value and descriptor to byte array blob

Reply:
@burakgok it looks fine now, `setJobMetrics` is invoked only for status change and after job is finished (also status change). I keep the comment open due to @Fly-Style's comments